src/scripts/main.js

```diff
@@ -1,3 +1,40 @@
 'use strict';
 
-// write code here
+const headers = document.querySelectorAll('th');
+const tBody = document.querySelector('tBody');
+const rows = document.querySelectorAll('tBody tr');
+
+const arrayOf = [...rows];
+
+headers.forEach((header) => {
+  header.addEventListener('click', function () {
+    const headerIndex = event.target.cellIndex;
+
+    sortBy(headerIndex);
+  });
+});
+
+function sortBy(headerIndex) {
+  arrayOf.sort((td1, td2) => {
```

Comment:
Your sort function is currently relying on side effects by directly modifying the `arrayOf` variable. It would be better to make your functions pure by not having them modify variables outside of their scope. You could achieve this by having your `sortBy` function take in the array to sort as an argument.